core/src/context.rs

```diff
@@ -121,6 +61,34 @@ impl Context {
         100
     }
 
+    // test_check_conductor_api() is used to inject a conductor_api with a working
+    // mock of agent/sign to be used in tests.
+    // There are two different implementations of this function below which get pulled
+    // in depending on if "test" is in the build config, or not.
+    // This allows unit tests of core to not have to deal with a conductor_api.
+    #[cfg(not(test))]
+    fn test_check_conductor_api(
+        conductor_api: Option<Arc<RwLock<IoHandler>>>,
+        _agent_id: AgentId,
+    ) -> Arc<RwLock<IoHandler>> {
+        // If you get here through this panic make sure that the context passed into the instance
+        // gets created with a real container API. In test config it will be populated with mock API
+        // that implements agent/sign with the mock_signer. We need this for testing but should
+        // never use that code in production!
+        // Hence the two different cases here.
+        conductor_api.expect("Context can't be created without conductor API")
+    }
+
+    #[cfg(test)]
+    fn test_check_conductor_api(
+        conductor_api: Option<Arc<RwLock<IoHandler>>>,
+        agent_id: AgentId,
+    ) -> Arc<RwLock<IoHandler>> {
+        conductor_api
+            .or(Some(Arc::new(RwLock::new(mock_conductor_api(agent_id)))))
```

Comment:
```suggestion
            .unwrap_or_else(|| Arc::new(RwLock::new(mock_conductor_api(agent_id))))
```

Reply:
Option::or returns an option. so would need to be
```suggestion
            .or(Some(Arc::new(RwLock::new(mock_conductor_api(agent_id))))).unwrap()
```
Maybe a matter of style, but Isn't `unwrap_or_else()` better here?

Reply:
sorry, misread the change request. yes, your suggestion is much better :)